stablehlo/tests/InterpreterDialect.td

```diff
@@ -0,0 +1,28 @@
+/* Copyright 2023 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_TESTS_INTERPRETER_DIALECT
+#define STABLEHLO_TESTS_INTERPRETER_DIALECT
+
+def Interpreter_Dialect : Dialect {
+  let name = "interpreter";
+  let cppNamespace = "::mlir::stablehlo::interpreter";
+
+  let description = [{
+    Interpreter dialect to implement StableHLO interpreter-related tests.
```

Comment:
Also, given that let's move the dialect from stablehlo/tests to stablehlo/reference?